### PR TITLE
Add tests and dom emulation

### LINKED ISSUE #12: ReferenceError: document is not defined
```
doc = document,
			      ^
ReferenceError: document is not defined
```

Seems to be an issue when trying to run this using an isomorphic app

### LINKED ISSUE #27: Find a way to automatize testing

### PR DESCRIPTION
**Pull request by kirjs**
Add a simulateDom function, closes #12 
Add a simple test, closes #27

### CODE REVIEW

File: src/simulateDOM.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+var jsdom = require('jsdom');
+
+global.document = jsdom.jsdom('<!doctype html><html><body><div id="test"></div></body></html>');
+var win = global.document.defaultView;
+global.window = global;
+for( var i in win ){
+  if( i !== 'window' && win.hasOwnProperty(i)){
+    global.window[i] = win[i];
```
Comment (aldendaniels) [inline src/simulateDOM.js]: Wondering if there's a way to do this without stuffing the global namespace full `JSDOM` variables.

Could we use WebPack to build a modified version of `Highcharts.js` that prepends the following:

```javascript
var jsdom = require('jsdom');
var document = jsdom.jsdom('<!doctype html><html><body><div id="test"></div></body></html>');
var window = document.defaultView;
// ... rest of Highcarts.js here
```

Comment (kirjs) [inline src/simulateDOM.js]: To my understanding it's not even possible to use jsdom with webpack (see #12), but I haven't try too hard, so feel free to experiment.

I'm definitely open to nicer solutions here.

Comment (aldendaniels) [inline src/simulateDOM.js]: My idea was to (during the build process) generate `dist/HighchartsNode.js`, which would shim only its local scope with `JSDOM`, without leaking into the global scope. In principle, there's no reason this shouldn't work. No problem though if you don't want to mess with this further. At a minimum though, I'd suggest adding a warning letting users know that `SimulateDOM` will pollute the global scope.

Comment (kirjs) [inline src/simulateDOM.js]: This makes sense, I added #31

Comment (aldendaniels) [inline src/simulateDOM.js]: :+1:

**Pull request merged**